### PR TITLE
Accessibility: Improve keyboard accessibility in `Collapse`

### LINKED ISSUE #59021: A11y: Fix keyboard accessibility in `packages/grafana-ui/src/components/Collapse/Collapse.tsx`

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- uses the correct semantic element (`<button>`) for the collapse header
- applies some styling to make sure it still looks the same

**Why do we need this feature?**

- better keyboard accessibility across grafana

**Who is this feature for?**

- everyone! 🙌 

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/59021

**Special notes for your reviewer**:

### CODE REVIEW

File: packages/grafana-ui/src/components/Collapse/Collapse.tsx [hunk 1]
```diff
@@ -145,10 +145,10 @@ export const Collapse = ({
 
   return (
     <div className={panelClass}>
-      <div className={headerClass} onClick={onClickToggle}>
+      <button type="button" className={cx(buttonStyles, headerClass)} onClick={onClickToggle}>
```
Comment (Clarity-89) [inline packages/grafana-ui/src/components/Collapse/Collapse.tsx:148]: Seems like a similar button is used more than once, should we have a separate component for it? Maybe a modified version of `IconButton`?

**Review by Clarity-89 — APPROVED** [review]

File: packages/grafana-ui/src/components/Collapse/Collapse.tsx (hunk 1 above)
Comment (ashharrison90) [reply packages/grafana-ui/src/components/Collapse/Collapse.tsx:148]: @Clarity-89 don't quite understand - do you mean creating a component for `<button type="button ...>`? 🤔

Comment (Clarity-89) [reply packages/grafana-ui/src/components/Collapse/Collapse.tsx:148]: Basically, but also with the clear button styles.

Comment (ashharrison90) [reply packages/grafana-ui/src/components/Collapse/Collapse.tsx:148]: ah got it. yeah that could have value... 🤔  lemme have a think about implementation

Comment (ashharrison90) [reply packages/grafana-ui/src/components/Collapse/Collapse.tsx:148]: gonna merge for now whilst we think about an implementation for exposing an unstyled button.

debating between just exposing a new `Button` variant (`unstyled`), or a whole new component `UnstyledButton`. neither of which feel super nice tbh. wondering whether we _shouldn't_ expose this as although it's being used quite a bit internally, i don't think we'd want to expose it externally as part of `@grafana/ui`? idk, any thoughts? 🤔

**Pull request merged**